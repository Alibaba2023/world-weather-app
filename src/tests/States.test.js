import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import State from '../components/States';

// Sample test data
const stateData = {
  countryState: 'Sample State',
  icon: 'https://example.com/sample-icon.png',
  date: '2023-08-03',
  weatherCondition: 'Sunny',
  temperature: 25,
};

describe('State component', () => {
  it('renders the state name, weather details, and compares to Snapshot', () => {
    const statesComponent = render(
      <MemoryRouter>
        <State
          countryState={stateData.countryState}
          icon={stateData.icon}
          date={stateData.date}
          weatherCondition={stateData.weatherCondition}
          temperature={stateData.temperature}
        />
      </MemoryRouter>,
    );
    expect(statesComponent).toMatchSnapshot();
  });
});
