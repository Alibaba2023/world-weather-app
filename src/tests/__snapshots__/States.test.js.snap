// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`State component renders the state name, weather details, and compares to Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="weather-card"
      >
        <div
          class="one"
        >
          <h3
            class="state-name"
          >
            Sample State
          </h3>
          <h2
            class="weather-date"
          >
            2023-08-03
          </h2>
        </div>
        <div
          class="2"
        >
          <h3
            class="weather-temp"
          >
            25
            °C
          </h3>
        </div>
        <div
          class="box-3"
        >
          <img
            alt="weatherImage"
            class="weather-image"
            src="https://example.com/sample-icon.png"
          />
          <p
            class="weather-condition"
          >
            Sunny
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="weather-card"
    >
      <div
        class="one"
      >
        <h3
          class="state-name"
        >
          Sample State
        </h3>
        <h2
          class="weather-date"
        >
          2023-08-03
        </h2>
      </div>
      <div
        class="2"
      >
        <h3
          class="weather-temp"
        >
          25
          °C
        </h3>
      </div>
      <div
        class="box-3"
      >
        <img
          alt="weatherImage"
          class="weather-image"
          src="https://example.com/sample-icon.png"
        />
        <p
          class="weather-condition"
        >
          Sunny
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
